Sources/SymbolKit/UnifiedSymbolGraph/UnifiedSymbolGraph.swift

```diff
@@ -160,6 +160,190 @@ extension UnifiedSymbolGraph {
+                    siblingOverloadGroups.insert(processingOverloadGroup)
+
+                    for overloadedSymbol in overloadGroups[processingOverloadGroup]! {
+                        overloadedSymbols.insert(overloadedSymbol)
+                        for otherOverloadGroup in overloadGroupsBySymbol[overloadedSymbol]! {
+                            if !siblingOverloadGroups.contains(otherOverloadGroup),
+                               !pendingOverloadGroups.contains(otherOverloadGroup)
+                            {
+                                pendingOverloadGroups.insert(otherOverloadGroup)
+                            }
+                        }
+                    }
+                }
+            }
+
+            guard !overloadedSymbols.isEmpty else { continue }
+
+            // 2. Sort the overloaded symbols with the same comparator as `SymbolGraph.createOverloadGroupSymbols()`.
+            var allSymbolsHaveDeclaration = true
+            let sortedOverloads = overloadedSymbols.map({ identifier in
+                let symbol = symbols[identifier]!
+                if !symbol.mixins.values.contains(where: { $0.keys.contains(SymbolGraph.Symbol.DeclarationFragments.mixinKey) }) {
+                    allSymbolsHaveDeclaration = false
+                }
+                return symbol
+            }).sorted(by: Symbol.sortForOverloads(orderByDeclaration: allSymbolsHaveDeclaration))
+            let firstOverload = sortedOverloads.first!
+            let computedOverloadGroup = firstOverload.uniqueIdentifier + SymbolGraph.Symbol.overloadGroupIdentifierSuffix
+
+            guard siblingOverloadGroups.contains(computedOverloadGroup) else {
+                assertionFailure(
+                """
+                UnifiedSymbolGraph.combineOverloadGroups found a candidate overload group that didn't
+                match any existing overload group from its constituent symbol graphs.
+
+                The unified graph found that \(firstOverload.uniqueIdentifier) should be the default
+                overload, but the constituent symbol graphs contained the following overload groups:
+                \(siblingOverloadGroups)
+                """
+                )
+                continue
+            }
+
+            for overloadIndex in sortedOverloads.indices {
+                let overloadedSymbol = sortedOverloads[overloadIndex]
+
+                // 3. Ensure that all the overloaded symbols have `overloadOf` relationships pointing
+                // to the computed overload group.
+                for selector in overloadedSymbol.mainGraphSelectors {
+                    let newRelationship = SymbolGraph.Relationship(
+                        source: overloadedSymbol.uniqueIdentifier,
+                        target: computedOverloadGroup,
+                        kind: .overloadOf,
+                        targetFallback: nil)
+                    if relationshipsByLanguage[selector]?.contains(where: { otherRelationship in
```

Comment:
Any performance concerns with this lookup? Should we be using a set instead of an array for `relationshipsByLanguage`'s values?

Reply:
`relationshipsByLanguage` is public API, so changing its type is a breaking change. 😢 The relationships listing can get pretty large, though, so there might be a better way to perform this lookup so that we don't potentially iterate over thousands of items dozens of times. I'll have to take a look.